### PR TITLE
Metamask Bug

### PR DESCRIPTION
**Pull request by techeng322**

### CODE REVIEW

File: hooks/useTBAPurchase.tsx [hunk 1]
```diff
@@ -22,8 +26,9 @@ const useTBAPurchase = () => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [])
 
-  const purchase = async (quantity: number) => {
+  const purchase = async (quantity: number, hasWallet = null) => {
```
Comment (sweetmantech) [inline hooks/useTBAPurchase.tsx]: why do we need to pass `hasWallet` as a param?

`useTbaPurchase` has access to `useUserProvider` as well right?

File: components/Pages/LandingPage/CounterDown.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const CounterDown = () => (
```
Comment (sweetmantech) [inline components/Pages/LandingPage/CounterDown.tsx]: refactor this component
`<Presave />`

File: components/Pages/LandingPage/LandingPage.tsx [hunk 3]
```diff
@@ -1,34 +1,21 @@
...
-        <div className="md:col-span-2 w-full">
-          <CollectButtons />
-        </div>
-        <div className="md:col-span-2 flex justify-center">
-          <Bundles />
+    <LiveHome />
+    {/* {screenMode === SCREEN.COLLECT && <LiveHome />} */}
+    {/* {screenMode === SCREEN.COUNTDOWN && (
+        <div className="w-[100%] h-[100vh] flex justify-center items-center">
+          <CounterDown />
```
Comment (sweetmantech) [inline components/Pages/LandingPage/LandingPage.tsx]: remove unused code 🚮

File: hooks/useTBAPurchase.tsx (hunk 1 above)
Comment (techeng322) [reply hooks/useTBAPurchase.tsx]: fixed ✅

File: components/Pages/LandingPage/LandingPage.tsx (hunk 3 above)
Comment (techeng322) [reply components/Pages/LandingPage/LandingPage.tsx]: fixed ✅

File: components/Pages/LandingPage/CounterDown.tsx (hunk 2 above)
Comment (techeng322) [reply components/Pages/LandingPage/CounterDown.tsx]: fixed ✅

File: hooks/useConnectedWallet.tsx [hunk 4]
```diff
@@ -1,9 +1,13 @@
 import { useWallets } from "@privy-io/react-auth"
+import { useMemo } from "react"
 
 const useConnectedWallet = () => {
   const { wallets } = useWallets()
   const privyWallet = wallets?.find((wallet) => wallet.walletClientType === "privy")
-  const metamaskWallet = wallets?.find((wallet) => wallet.walletClientType === "metamask")
+  const metamaskWallet = useMemo(
+    () => wallets?.find((wallet) => wallet.walletClientType === "metamask"),
```
Comment (sweetmantech) [inline hooks/useConnectedWallet.tsx]: this needs to be more `generic`

needs to work with `any` `external` wallet, not just metamask

**Review by sweetmantech — APPROVED** [review]

**Pull request merged**